var/spack/repos/builtin/packages/python/package.py

```diff
@@ -360,710 +331,470 @@ def flag_handler(self, name, flags):
+        Returns:
+            str: platform-independent site-packages directory
+        """
+        prefix = self.config_vars["base"] + os.sep
+        path = self.config_vars["purelib"]
+        if path.startswith(prefix):
+            return path.replace(prefix, "")
+        return os.path.join("lib", "python{}".format(self.version.up_to(2)), "site-packages")
 
-            # Ensure that ctypes module works
-            if "+ctypes" in spec:
-                self.command("-c", "import ctypes")
+    @property
+    def include(self):
+        """Directory for non-platform-specific header files.
 
-            # Ensure that tkinter module works
-            # https://wiki.python.org/moin/TkInter
-            if "+tkinter" in spec:
-                # Only works if ForwardX11Trusted is enabled, i.e. `ssh -Y`
-                if "DISPLAY" in env:
-                    self.command("-c", "import tkinter; tkinter._test()")
-                else:
-                    self.command("-c", "import tkinter")
+        Exact directory depends on platform/Python version/ABI flags. Examples include:
 
-            # Ensure that uuid module works
-            if "+uuid" in spec:
-                self.command("-c", "import uuid")
+        * ``include/pythonX.Y`` on most POSIX systems
+        * ``include/pythonX.Yd`` for debug builds
+        * ``include/pythonX.Ym`` for malloc builds
+        * ``include/pythonX.Yu`` for wide unicode builds
+        * ``include`` on macOS with framework Python
+        * ``Include`` on Windows
 
-            # Ensure that tix module works
-            if "+tix" in spec:
-                self.command("-c", "import tkinter.tix")
-
-            # Ensure that crypt module works
-            if "+crypt" in spec:
-                self.command("-c", "import crypt")
-
-    # ========================================================================
-    # Set up environment to make install easy for python extensions.
-    # ========================================================================
-
-    @property
-    def command(self):
-        """Returns the Python command, which may vary depending
-        on the version of Python and how it was installed.
+        Returns:
+            str: platform-independent header file directory
+        """
+        prefix = self.config_vars["installed_base"] + os.sep
+        path = self.config_vars["include"]
+        if path.startswith(prefix):
+            return path.replace(prefix, "")
+        return os.path.join("include", "python{}".format(self.version.up_to(2)))
 
-        In general, Python 3 only comes with a ``python3`` command. However, some
-        package managers will symlink ``python`` to ``python3``, while others
-        may contain ``python3.11``, ``python3.10``, and ``python3.9`` in the
-        same directory.
+    def setup_run_environment(self, env):
+        env.prepend_path("CPATH", os.pathsep.join(self.spec["python"].headers.directories))
```

Comment:
Considering `CPATH` isn't a consideration on Windows, I'm almost 100% sure I did not make this change (and I can't remember making it either), it's likely rebase/maintenance cruft since this change is so old. Removed.